pkg/adapter/nimbus-kyverno/manager/manager.go

```diff
@@ -0,0 +1,363 @@
+		return
+	}
+
+	var kpsOwnedByNp []kyvernov1.Policy
+	for _, kp := range existingkps.Items {
+		ownerRef := kp.OwnerReferences[0]
+		if ownerRef.Name == np.Name && ownerRef.UID == np.UID {
+			kpsOwnedByNp = append(kpsOwnedByNp, kp)
+		}
+	}
+	if len(kpsOwnedByNp) == 0 {
+		return
+	}
+
+	kpsToDelete := make(map[string]kyvernov1.Policy)
+
+	// Populate owned kps
+	for _, kpOwnedByNp := range kpsOwnedByNp {
+		kpsToDelete[kpOwnedByNp.Name] = kpOwnedByNp
+	}
+
+	for _, nimbusRule := range np.Spec.NimbusRules {
+		kpName := np.Name + "-" + strings.ToLower(nimbusRule.ID)
+		delete(kpsToDelete, kpName)
+	}
+
+	for kpName := range kpsToDelete {
+		kp := kpsToDelete[kpName]
+		if err := k8sClient.Delete(ctx, &kp); err != nil {
+			logger.Error(err, "failed to delete dangling KyvernoPolicy", "KyvernoPolicy.Name", kp.Namespace, "KyvernoPolicy.Namespace", kp.Namespace)
+			continue
+		}
+
+		logger.Info("Dangling KyvernoPolicy deleted", "KyvernoPolicy.Name", kp.Name, "KyvernoPolicy.Namespace", kp.Namespace)
+	}
+}
+
+func deleteKcp(ctx context.Context, cnpName string) {
+	logger := log.FromContext(ctx)
+	var kcps kyvernov1.ClusterPolicyList
+
+	if err := k8sClient.List(ctx, &kcps); err != nil {
+		logger.Error(err, "failed to list KyvernoClusterPolicies")
+		return
+	}
+
+	// Kubernetes GC automatically deletes the child when the parent/owner is
+	// deleted. So, we don't need to do anything in this case since NimbusPolicy is
+	// the owner and when it gets deleted corresponding kps will be automatically
+	// deleted.
+	for _, kcp := range kcps.Items {
+		logger.Info("KyvernoClusterPolicy already deleted due to ClusterNimbusPolicy deletion",
+			"KyvernoClusterPolicy.Name", kcp.Name, 
+			"ClusterNimbusPolicy.Name", cnpName,
+		)
+	}
+}
+
+func deleteDanglingkcps(ctx context.Context, cnp intentv1.ClusterNimbusPolicy, logger logr.Logger) {
+	var existingkcps kyvernov1.ClusterPolicyList
+	if err := k8sClient.List(ctx, &existingkcps); err != nil {
+		logger.Error(err, "failed to list KyvernoClusterPolicies for cleanup")
+		return
+	}
+
+	var kcpsOwnedByCnp []kyvernov1.ClusterPolicy
+	for _, kcp := range existingkcps.Items {
+		ownerRef := kcp.OwnerReferences[0]
+		if ownerRef.Name == cnp.Name && ownerRef.UID == cnp.UID {
+			kcpsOwnedByCnp = append(kcpsOwnedByCnp, kcp)
+		}
+	}
+	if len(kcpsOwnedByCnp) == 0 {
+		return
+	}
+
+	kcpsToDelete := make(map[string]kyvernov1.ClusterPolicy)
+
+	// Populate owned kcps
+	for _, kcpOwnedByCnp := range kcpsOwnedByCnp {
+		kcpsToDelete[kcpOwnedByCnp.Name] = kcpOwnedByCnp
+	}
+
+	for _, nimbusRule := range cnp.Spec.NimbusRules {
+		kcpName := cnp.Name + "-" + strings.ToLower(nimbusRule.ID)
+		delete(kcpsToDelete, kcpName)
+	}
+
+	for kcpName := range kcpsToDelete {
+		kcp := kcpsToDelete[kcpName]
+		if err := k8sClient.Delete(ctx, &kcp); err != nil {
+			logger.Error(err, "failed to delete dangling KyvernoClusterPolicy", "KyvernoClusterPolicy.Name", kcp.Name)
+			continue
+		}
+
```

Comment:
Add a todo/reminder to remember to call the clusternimbuspolicy update status API in a follow-up PR to update the status subresource.